### PR TITLE
EVG-3476: spawn only containers in scheduler when working on static hosts

### PR DESCRIPTION
**Pull request by li-cindy**

### CODE REVIEW

File: scheduler/scheduler.go [hunk 1]
```diff
@@ -296,6 +296,10 @@ func parentCapacity(d distro.Distro, numNewParents, numCurrentParents, numCurren
 	if numNewParents+numCurrentParents > d.PoolSize {
 		numNewParents = d.MaxContainers - numCurrentParents
 	}
+	// if looking at a static docker provider, do not spawn any more parents
```
Comment (john-m-liu) [inline scheduler/scheduler.go]: I would move this to the top of the function since nothing else is relevant for static parents

Comment (li-cindy) [reply scheduler/scheduler.go]: Actually this still would mean the `if numNewParents+numCurrentParents > d.PoolSize` block will still run before returning at the end. Should I just immediately return after each if statement instead?

Comment (li-cindy) [reply scheduler/scheduler.go]: Or use `else if`s? I don't know if we use those often

Comment (john-m-liu) [reply scheduler/scheduler.go]: I would move this to the top and return 0 immediately. I think that's the simplest

**Pull request merged**